MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```
@@ -32,340 +30,296 @@ namespace MediaBrowser.MediaEncoding.Encoder
-            }
-
-            var newPaths = GetEncoderPaths(appPath);
-            if (string.IsNullOrWhiteSpace(newPaths.Item1) || string.IsNullOrWhiteSpace(newPaths.Item2) || IsSystemInstalledPath(appPath))
+            if (Directory.Exists(path))
             {
-                newPaths = TestForInstalledVersions();
+                return GetEncoderPathFromDirectory(path, "ffmpeg");
             }
 
-            if (!string.IsNullOrWhiteSpace(newPaths.Item1) && !string.IsNullOrWhiteSpace(newPaths.Item2))
+            if (File.Exists(path))
             {
-                FFMpegPath = newPaths.Item1;
-                FFProbePath = newPaths.Item2;
+                return path;
             }
 
-            LogPaths();
+            return null;
         }
 
-        private Tuple<string, string> GetEncoderPaths(string configuredPath)
+        private string GetEncoderPathFromDirectory(string path, string filename)
         {
-            var appPath = configuredPath;
-
-            if (!string.IsNullOrWhiteSpace(appPath))
+            try
             {
-                if (Directory.Exists(appPath))
-                {
-                    return GetPathsFromDirectory(appPath);
-                }
+                var files = FileSystem.GetFilePaths(path);
 
-                if (File.Exists(appPath))
-                {
-                    return new Tuple<string, string>(appPath, GetProbePathFromEncoderPath(appPath));
-                }
-            }
+                var excludeExtensions = new[] { ".c" };
 
-            return new Tuple<string, string>(null, null);
+                return files.FirstOrDefault(i => string.Equals(Path.GetFileNameWithoutExtension(i), filename, StringComparison.OrdinalIgnoreCase)
+                                                    && !excludeExtensions.Contains(Path.GetExtension(i) ?? string.Empty));
+            }
+            catch (Exception)
```

Do checks instead of silently failing.

---

Think this is ok as not interested in 'why' it may fail.  Just return null under exception.